project/Subject.cs

```diff
@@ -1,152 +1,169 @@
 namespace assignment_3
 {
-    public class Subject
-    {
-        public string SubjectId { get; private set; }
+public class Subject
+{
+    public string SubjectId { get; private set; }
 
-        private string _subjectName;
-        public string SubjectName
+    private string _subjectName;
+    public string SubjectName
+    {
+        get => _subjectName;
+        private set
         {
-            get => _subjectName;
-            private set
-            {
-                if (string.IsNullOrWhiteSpace(value))
-                    throw new ArgumentException("Subject name cannot be null or empty.");
-                _subjectName = value;
-            }
+            if (string.IsNullOrWhiteSpace(value))
+                throw new ArgumentException("Subject name cannot be null or empty.");
+            _subjectName = value;
         }
+    }
 
-        private int _gradingScale;
-        public int GradingScale
+    private int _gradingScale;
+    public int GradingScale
+    {
+        get => _gradingScale;
+        private set
         {
-            get => _gradingScale;
-            private set
-            {
-                if (value < 1 || value > 5)
-                    throw new ArgumentException("Grading scale must be between 1 and 5.");
-                _gradingScale = value;
-            }
+            if (value < 1 || value > 5)
+                throw new ArgumentException("Grading scale must be between 1 and 5.");
+            _gradingScale = value;
         }
+    }
 
-        private List<Teacher> _teachers = new();
-        public List<Teacher> Teachers
+    private List<Teacher> _teachers = new();
+    public List<Teacher> Teachers
+    {
+        get => new(_teachers);
+        set
         {
-            get => new(_teachers);
-            set
```

Comment:
These validation properties must not be in here, they need to be in the CRUD operations.